metals/src/main/scala/scala/meta/internal/decorations/ImplicitDecorationProvider.scala

```diff
@@ -0,0 +1,368 @@
+package scala.meta.internal.decorations
+
+import java.nio.charset.Charset
+import java.nio.file.Path
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.ClientConfiguration
+import scala.meta.internal.metals.Diagnostics
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.UserConfiguration
+import scala.meta.internal.metap.PrinterSymtab
+import scala.meta.internal.mtags.Md5Fingerprints
+import scala.meta.internal.mtags.SemanticdbClasspath
+import scala.meta.internal.mtags.Semanticdbs
+import scala.meta.internal.semanticdb.Print
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.internal.semanticdb.TextDocuments
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.io.AbsolutePath
+import scala.meta.metap.Format
+
+import org.eclipse.{lsp4j => l}
+
+final class ImplicitDecorationProvider(
```

Comment:
About the name, does this only print implicit values? Doesn't it also print inferred type parameters and other synthetics?

Reply:
Renamed to `SyntheticsDecorationProvider`